src/parser/helper/QueryStatements.ts

```diff
@@ -0,0 +1,117 @@
+import { Languages, languageToAbbreviation } from "./Languages";
+
+export interface QueryStatementInterface {
+    activatedFor(language: Languages): boolean;
+
+    /**
+     * Tests whether this query is grammatically applicable for the specified programming language.
+     * @param language The language to check.
+     * @return true if the query is applicable, false if not.
+     */
+    applicableFor(language: Languages): boolean;
+
+    toQuery(): string;
+}
+
+export class SimpleQueryStatement implements QueryStatementInterface {
+    query: string;
+
+    constructor(query: string) {
+        this.query = query;
+    }
+
+    activatedFor(language: Languages): boolean {
+        return true;
+    }
+
+    applicableFor(language: Languages): boolean {
+        return true;
+    }
+
+    toQuery(): string {
+        return this.query;
+    }
+}
+
+export abstract class LanguageSpecificQueryStatement implements QueryStatementInterface {
+    protected activatedForLanguages: Set<Languages>;
+    protected applicableForLanguages: Set<Languages>;
+
+    protected constructor(applicable_for_languages: string[], activated_for_languages?: string[]) {
+        this.applicableForLanguages =
+            languageToAbbreviation.getKeysForAllValues(applicable_for_languages);
+
+        if (activated_for_languages !== undefined) {
+            this.activatedForLanguages =
+                languageToAbbreviation.getKeysForAllValues(activated_for_languages);
+        } else {
+            this.activatedForLanguages = new Set();
+        }
+    }
+
+    activatedFor(language: Languages): boolean {
+        return this.activatedForLanguages.size === 0 || this.activatedForLanguages.has(language);
+    }
+
+    applicableFor(language: Languages): boolean {
+        return this.applicableForLanguages.has(language);
+    }
+
+    abstract toQuery(): string;
+}
+
+export class ExpressionQueryStatement extends LanguageSpecificQueryStatement {
+    private readonly expression: string;
```

Comment:
I suggest to use `#expression` instead of `private expression`. It's the way in JS to make a variable actually private. The keyword private is used in TS but it's just a fake private thing and it's still accessible from the outside. I believe we also do not have to mark it as readonly, when using a private variable.

Reply:
Ah, good to know, I will change all private fields to # then.
That `readonly` thing is a recommendation from the IDE that still pops up when using #. I guess that making that field readonly to also prevent modification from inside the class does not do any harm.